Improve keyboard accessibility for buddy list and stream list.

Lalit3716 opened the pull request:
Improve keyboard accessibility for buddy list as discussed on [CZO](https://chat.zulip.org/#narrow/stream/137-feedback/topic/keyboard.20accessibility)

The general process for accessing vdot menu will be:
1. Hit W to pop up the "who" menu.
2. Navigate using up/down arrow key without losing focus from search input.
3. Tab to highlight the vdot menu for the selected row.
4. Hit "Enter" key to open up the menu.
5. Use arrows to navigate the menu.

Fixes: #25391 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! When I was testing this PR, I was thinking that you want to tab to the vdot menu, and then press "Enter" to actually open the use card. Looks like that was the design proposed on the CZO thread as well, so let's try that.

Lalit3716 committed:
refactor: Refactor `list_cursor` to actually focus rows for better keyboard access.

This commit refactors the `list_cursor` such that we don't stimulate the focus on rows
anymore but actually focus them for better keyboard accessibility.

I achieved this by giving `tabindex` value of 0 to `presence_row` and then making a function
`handle_navigation` in `list_cursor` which sets up all the neccessary handlers for better
keyboard navigation.

Lalit3716 committed:
list_cursor: Make vdot menu icon keyboard accessible.

Set `tabindex` to 0 on `user-list-sidebar-menu-icon` to make it focusable from
keyboard. Used `:focus-within` selector in css to apply styles on row even if
vdot icon is in focus and not the row itself.

Also added a `Tab` handlers to actually focus the vdot menu icon.

Fixes #25391.

Lalit3716 commented:
@alya It is ready for another look from you, I just want to make sure if that's the keyboard interaction we want before trying to fix tests because they look a little bit harder to fix 😕

alya commented:
I haven't done any testing on the right sidebar in this pass, but I some of the above comments may apply there as well.

Lalit3716 commented:
Thanks for a thorough review on this Alya! I will try to fix those issues.
